### PR TITLE
Add LFI exploit prevention support

### PR DESCRIPTION
**Pull request by jandro996**
# What Does This Do

- Gives support to LFI exploit prevention by updating IAST Path traversal callsites 

# Motivation

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [APPSEC-46822](https://datadoghq.atlassian.net/browse/APPSEC-46822)

[APPSEC-46822]: https://datadoghq.atlassian.net/browse/APPSEC-46822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by smola — CHANGES_REQUESTED** [review]
Needs to announce RC capability.

File: dd-java-agent/instrumentation/java-io/src/test/groovy/datadog/trace/instrumentation/java/io/FileLoadedRaspHelperForkedTest.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    when:
+    FileLoadedRaspHelper.INSTANCE."onFileLoaded"(*args)
+
+    then:
+    1 * callbackProvider.getCallback(EVENTS.fileLoaded()) >> listener
+    1 * listener.apply(reqCtx, expected) >> flow
+
+    where:
+    args                                                      |  expected
```
Comment (smola) [inline dd-java-agent/instrumentation/java-io/src/test/groovy/datadog/trace/instrumentation/java/io/FileLoadedRaspHelperForkedTest.groovy:29]: Can  you format this block?

File: dd-java-agent/appsec/src/main/java/com/datadog/appsec/gateway/GatewayBridge.java [hunk 2]
```diff
@@ -159,6 +161,31 @@ private Flow<Void> onNetworkConnection(RequestContext ctx_, String url) {
...
+        subInfo = producerService.getDataSubscribers(KnownAddresses.IO_FS_FILE);
+        ioFileSubInfo = subInfo;
+      }
+      if (subInfo == null || subInfo.isEmpty()) {
+        return NoopFlow.INSTANCE;
+      }
+      DataBundle bundle =
+          new MapDataBundle.Builder(CAPACITY_0_2).add(KnownAddresses.IO_FS_FILE, path).build();
+      try {
+        GatewayContext gwCtx = new GatewayContext(false, RuleType.LFI);
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/appsec/src/main/java/com/datadog/appsec/gateway/GatewayBridge.java:181]:
```suggestion
        GatewayContext gwCtx = new GatewayContext(true, RuleType.LFI);
```
All RASP operations should be transient

File: internal-api/src/main/java/datadog/trace/api/telemetry/RuleType.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package datadog.trace.api.telemetry;
 
 public enum RuleType {
-  LIF("lfi"),
+  LFI("lfi"),
```
Comment (manuel-alvarez-alvarez) [inline internal-api/src/main/java/datadog/trace/api/telemetry/RuleType.java:4]: :laughing:

File: internal-api/src/main/java/datadog/trace/api/gateway/Events.java [hunk 4]
```diff
@@ -248,6 +248,17 @@ public EventType<BiFunction<RequestContext, String, Flow<Void>>> networkConnecti
     return (EventType<BiFunction<RequestContext, String, Flow<Void>>>) NETWORK_CONNECTION;
   }
 
+  static final int FILE_LOADED_ID = 20;
```
Comment (manuel-alvarez-alvarez) [inline internal-api/src/main/java/datadog/trace/api/gateway/Events.java:251]: We might have some conflicts here with my [PR](https://github.com/DataDog/dd-trace-java/pull/7591/files#diff-417adb6f8aaf085942b7ea7417a403b0c526bf5ef8e21efddb4aaa0fb2eb2b44)

File: dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/gateway/GatewayBridgeSpecification.groovy [hunk 5]
```diff
@@ -798,6 +800,26 @@ class GatewayBridgeSpecification extends DDSpecification {
...
+    when:
+    Flow<?> flow = fileLoadedCB.apply(ctx, path)
+
+    then:
+    1 * eventDispatcher.publishDataEvent(nonEmptyDsInfo, ctx.data, _ as DataBundle, _ as GatewayContext) >>
+    { a, b, db, gw -> bundle = db; gatewayContext = gw; NoopFlow.INSTANCE }
+    bundle.get(KnownAddresses.IO_FS_FILE) == path
+    flow.result == null
+    flow.action == Flow.Action.Noop.INSTANCE
+    gatewayContext.isTransient == false
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/appsec/src/test/groovy/com/datadog/appsec/gateway/GatewayBridgeSpecification.groovy:819]:
```suggestion
    gatewayContext.isTransient == true
```

**Review by ValentinZakharov — APPROVED** [review]

**Review by smola — APPROVED** [review]

**Review by manuel-alvarez-alvarez — APPROVED** [review]

**Pull request merged**